### ISSUE TITLE
SNOW-981509: Overwriting variable in global scope

### ISSUE THREAD

**Issue by pharturs** [snowflakedb/snowflake-connector-nodejs#712]
Please answer these questions before submitting your issue. 
In order to accurately debug the issue this information is required. Thanks!

1. What version of NodeJS driver are you using?
1.9.1

   
2. What operating system and processor architecture are you using?
*nix
   
3. What version of NodeJS are you using?
(`node --version` and `npm --version`)
node v18.16.1
npm 9.5.1

4. What are the component versions in the environment (`npm list`)?
```
npm list
snowflake-sdk@1.9.1 
├── @aws-sdk/client-s3@3.462.0
├── @aws-sdk/types@3.460.0
├── @azure/storage-blob@12.17.0 overridden
├── @google-cloud/storage@6.12.0
├── @techteamer/ocsp@1.0.1
├── agent-base@6.0.2
├── asn1.js-rfc2560@5.0.1
├── asn1.js-rfc5280@3.0.0
├── asn1.js@5.4.1
├── async@3.2.5
├── axios@1.6.2
├── big-integer@1.6.52
├── bignumber.js@9.1.2
├── binascii@0.0.2
├── bn.js@5.2.1
├── browser-request@0.3.3
├── debug@3.2.7
├── eslint@8.54.0
├── expand-tilde@2.0.2
├── extend@3.0.2
├── fast-xml-parser@4.3.2
├── fastest-levenshtein@1.0.16
├── generic-pool@3.9.0
├── glob@7.2.3
├── https-proxy-agent@5.0.1
├── jsonwebtoken@9.0.2
├── mime-types@2.1.35
├── mkdirp@1.0.4
├── mocha@10.2.0
├── mock-require@3.0.3
├── moment-timezone@0.5.43
├── moment@2.29.4
├── nyc@15.1.0
├── open@7.4.2
├── python-struct@1.1.3
├── simple-lru-cache@0.0.2
├── test-console@2.0.0
├── tmp@0.2.1
├── uuid@8.3.2
└── winston@3.11.0
```

5.Server version:* E.g. 1.90.1
You may get the server version by running a query:
```
SELECT CURRENT_VERSION();
```
not relevant

6. What did you do?
define a global `request` variable with a value, and observe it is being overwritten after connection to Snowflake is established.
```
  const connection = snowflake.createConnection(conf);
  await connection.connectAsync();
```

7. What did you expect to see?
no impact to the global environment.

8. Can you set logging to DEBUG and collect the logs?
not relevant
   https://community.snowflake.com/s/article/How-to-generate-log-file-on-Snowflake-connectors

  e.g
  Add this to get standard output.

```
var snowflake = require('snowflake-sdk');
snowflake.configure(
{
  logLevel: 'trace'
});
```
not relevant
   
9. What is your Snowflake account identifier, if any? (Optional)
not relevant

**Comment by sfc-gh-dszmolka** [comment]
hi - thank you for raising this issue with us and especially already suggesting a solution! thank you for your contribution - we'll review the PR

**Comment by sfc-gh-dszmolka** [comment]
your PR is now merged and will be part of the next release. Thank you for your contribution!

**Comment by sfc-gh-dszmolka** [comment]
released with 1.9.2, closing
